src/lib/emitters.rs

```diff
@@ -85,6 +94,151 @@ pub fn write_json_to_stream<T: RenderJson, U: RenderJson>(mut stream: Box<Write>
         .map_err(|_| EmitterError::CouldNotWriteToFile)
 }
 
+// Map action types to terminal colours.
+fn build_colour_map() -> HashMap<ActionType, term::color::Color> {
+    let mut map: HashMap<ActionType, term::color::Color> = HashMap::new();
+    map.insert(ActionType::DELETE, term::color::BRIGHT_RED);
+    map.insert(ActionType::INSERT, term::color::BRIGHT_GREEN);
+    map.insert(ActionType::MOVE, term::color::BRIGHT_BLUE);
+    map.insert(ActionType::UPDATE, term::color::BRIGHT_YELLOW);
+    map
+}
+
+// Read file and return its contents or `ParseError`.
+fn read_file(path: &str) -> Result<String, EmitterError> {
+    let mut f = match File::open(path) {
+        Ok(r) => r,
+        Err(_) => return Err(EmitterError::CouldNotOpenFile),
+    };
+    let mut s = String::new();
+    f.read_to_string(&mut s).unwrap();
+    Ok(s)
+}
+
+/// Write diff to STDOUT, highlighting actions in the edit script.
+///
+/// This function prints the whole file, rather than just the relevant context.
+/// Patch format is not supported, and hunk headers (e.g. `@@ -1 +1 @@`) are
+/// not printed.
+///
+/// ## Panics
+/// If the terminal foreground cannot be changed or reset by the
+/// [term](https://crates.io/crates/term) crate, this function will panic.
+/// However, `term` is cross-platform so this failure case should be rare.
+pub fn write_colour_diff_to_stdout<T: Clone + Display + Eq>(store: &MappingStore<T>,
+                                                            script: &EditScript<T>,
+                                                            from_path: &str,
+                                                            to_path: &str)
+                                                            -> Result<(), EmitterError> {
+    let colours = build_colour_map();
+    // Turn edit script and mappings into hunks of related patches.
+    let mut from_patches: Vec<Patch> = vec![];
+    let mut to_patches: Vec<Patch> = vec![];
+    script.patchify(store, &mut from_patches, &mut to_patches);
+    // No patches in either AST, so input files must be identical.
+    if from_patches.is_empty() && to_patches.is_empty() {
+        return Ok(());
+    }
+    let to_hunks = hunkify(to_patches);
+    let from_file = read_file(from_path)?;
+    let to_file = read_file(to_path)?;
+    let mut ch = 0;
```

Comment:
This variable is defined a long from where it's used. Maybe put it nearer to line 168?